spot_micro_joy/scripts/spotMicroJoystickMove.py

```diff
@@ -0,0 +1,150 @@
+        self._ros_pub_walk_cmd = rospy.Publisher('/walk_cmd', Bool, queue_size=1)
+        self._ros_pub_stand_cmd = rospy.Publisher('/stand_cmd', Bool, queue_size=1)
+        self._ros_pub_idle_cmd = rospy.Publisher('/idle_cmd', Bool, queue_size=1)
+
+        rospy.loginfo("Joystick control node publishers corrrectly initialized")
+
+    def reset_all_motion_commands_to_zero(self):
+        '''Reset body motion cmd states to zero and publish zero value body motion commands'''
+
+        self._vel_cmd_msg.linear.x = 0
+        self._vel_cmd_msg.linear.y = 0
+        self._vel_cmd_msg.linear.z = 0
+        self._vel_cmd_msg.angular.x = 0
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._ros_pub_vel_cmd.publish(self._vel_cmd_msg)
+
+    def reset_all_angle_commands_to_zero(self):
+        '''Reset angle cmd states to zero and publish them'''
+
+        self._angle_cmd_msg.x = 0
+        self._angle_cmd_msg.y = 0
+        self._angle_cmd_msg.z = 0
+
+        self._ros_pub_angle_cmd.publish(self._angle_cmd_msg)
+
+    def on_joy(self, msg):
+        if msg.buttons[self.BUTTON_GREEN] == 1:
+            self._ros_pub_idle_cmd.publish(self._idle_event_cmd_msg)
+            rospy.loginfo('Idle command issued from joystick.')
+            self.mode = self.MODE_IDLE
+
+        if msg.buttons[self.BUTTON_YELLOW] == 1:
+            self._ros_pub_stand_cmd.publish(self._stand_event_cmd_msg)
+            rospy.loginfo('Stand command issued from joystick.')
+            self.mode = self.MODE_STAND
+
+        if msg.buttons[self.BUTTON_BLUE] == 1:
+            self.reset_all_angle_commands_to_zero()
+            rospy.loginfo('Entering joystick angle command mode.')
+            self.mode = self.MODE_ANGLE
+
+        if msg.buttons[self.BUTTON_RED] == 1:
+            self.reset_all_angle_commands_to_zero()
+            self._ros_pub_walk_cmd.publish(self._walk_event_cmd_msg)
+            rospy.loginfo('Entering joystick walk command mode.')
+            self.mode = self.MODE_WALK
+
+        if self.mode == self.MODE_ANGLE:
+            print('Cmd Values: phi: %1.3f deg, theta: %1.3f deg, psi: %1.3f deg ' \
+                  % (
+                  self._angle_cmd_msg.x * 180 / pi, self._angle_cmd_msg.y * 180 / pi, self._angle_cmd_msg.z * 180 / pi))
+
+            self._angle_cmd_msg.x = pi / 180 * msg.axes[self.AXES_ROLL] * 45 * -1
+            self._angle_cmd_msg.y = pi / 180 * msg.axes[self.AXES_PITCH] * 45 * -1
+            self._angle_cmd_msg.z = pi / 180 * msg.axes[self.AXES_YAW] * 45
+            self._ros_pub_angle_cmd.publish(self._angle_cmd_msg)
+
+        if self.mode == self.MODE_WALK:
+            self._vel_cmd_msg.linear.x = msg.axes[self.WALK_AXES_FORWARD] * 0.05
+            self._vel_cmd_msg.linear.y = msg.axes[self.WALK_AXES_STRAFE] * 0.05
+            self._vel_cmd_msg.angular.z = pi / 180 * msg.axes[self.AXES_YAW] * 15
```

Comment:
Why not utilize the two analog sticks of a PS4-like controller independently to control speed and angle commands? This way you could avoid designing state into your node.

In walk mode, angle commands are ignored by the motion control node, and similarly in stand mode, speed commands are ignored, so each joystick would only move the robot if the robot's in the proper state.

Reply:
because i need both sticks in both modes. i tried to keep same actions on the same axes ... 
i walk mode we have 3 axes: forward, stafe (sideway), and yaw (missing elevation :) )
in angle-mode we have 3 axes as well ... roll, pitch, yaw ... (missing elevation as well :) )
of cause we can use the digital cross, but ... don't like ...

Reply:
Oh yea, I forgot you need at least three axes to control x, y, and yaw.